cmd/clusterctl/pkg/client/repository/templates_client.go

```diff
@@ -61,7 +52,32 @@ func newTemplatesClient(provider config.Provider, version string, repository Rep
 
 // Get return the template for the flavor/bootstrap provider specified.
 // In case the template does not exists, an error is returned.
-func (f *templatesClient) Get(flavor, bootstrap string, options TemplateOptions) (Template, error) {
-	//TODO: implement in a follow up PR
-	return nil, nil
+func (f *templatesClient) Get(flavor, bootstrap, targetNamespace string) (Template, error) {
+	if bootstrap == "" {
+		return nil, errors.New("invalid arguments: please proved a bootstrap provide name")
+	}
+
+	if targetNamespace == "" {
+		return nil, errors.New("invalid arguments: please proved a targetNamespace")
+	}
+
+	// we are always reading templatesClient for a well know version, that usually is
+	// the version of the provider installed in the management cluster.
+	// ndr. "latest" does not apply for get.
+	version := f.version
+
+	// building template name according with following the convention:
+	// config[-{flavor_name}]-{bootstrap_provider_name}.yaml
+	name := "config"
+	if flavor != "" {
+		name = fmt.Sprintf("%s-%s", name, flavor)
+	}
+	name = fmt.Sprintf("%s-%s.yaml", name, bootstrap)
+
+	rawyaml, err := f.repository.GetFile(version, name)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to read %q from the repository for provider %q", name, f.provider.Name())
```

Comment:
Would adding the `version` to the error message be helpful as well? Or is it redundant?

Reply:
Not at all!